File: net/import-content-model/CreateTaxonomy.cs
// DocSection: import_model_create_taxonomy
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kentico.Kontent.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_PROJECT_ID>"
});

var response = await client.CreateTaxonomyGroupAsync(new TaxonomyGroupCreateModel
{
    Name = "Blogpost topic",
    Codename = "blog_topic",
    Terms = new TaxonomyTermCreateModel[]
    {
        new TaxonomyTermCreateModel
        {
            Name = "Sport",
            Codename = "sport",
            Terms = new TaxonomyTermCreateModel[]
            {
                new TaxonomyTermCreateModel
                {
                    Name = "Soccer",
                    ExternalId = "soccer",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
                new TaxonomyTermCreateModel
                {
                    Name = "Ice hockey",
                    ExternalId = "hockey",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
                new TaxonomyTermCreateModel
                {
                    Name = "Rugby",
                    ExternalId = "rugby",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
            }
        },
        new TaxonomyTermCreateModel
        {
            Name = "Technology stack",
            Codename = "tech",
            Terms = new TaxonomyTermCreateModel[]
            {
                new TaxonomyTermCreateModel
                {
                    Name = "Javascript",
                    ExternalId = "js",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
                new TaxonomyTermCreateModel
                {
                    Name = "C#",
                    ExternalId = "c",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
                new TaxonomyTermCreateModel
                {
                    Name = "MVC",
                    ExternalId = "mvc",
                    Terms = Enumerable.Empty<TaxonomyTermCreateModel>()
                },
            }
        },
    }
});
// EndDocSection

File: js/import-content-model/CreateTaxonomy.js
// DocSection: import_model_create_taxonomy
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
// Using ES6 syntax
import { ManagementClient } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

const response = client.addTaxonomy()
  .withData(
    {
      name: 'Blogpost topic',
      codename: 'blog_topic',
      terms: [
        {
          name: 'Sport',
          codename: 'sport',
          terms: [
            {
              name: 'Soccer',
              codename: 'soccer',
              terms: []
            },
            {
              name: 'Ice hockey',
              codename: 'hockey',
              terms: []
            },
            {
              name: 'Rugby',
              codename: 'rugby',
              terms: []
            }
          ]
        },
        {
          name: "Technology stack",
          codename: "tech",
          terms: [
            {
              name: "JavaScript",
              codename: "js",
              terms: []
            },
            {
              name: "C#",
              codename: "c",
              terms: []
            },
            {
              name: 'MVC',
              codename: 'mvc',
              terms: []
            }
          ]
        }
      ]
    }
  )
  .toPromise();
// EndDocSection
